grow/cache/podcache.py

```diff
@@ -105,9 +108,14 @@ def _write_json(self, path, obj):
         output = json.dumps(
             obj, cls=json_encoder.GrowJSONEncoder,
             sort_keys=True, indent=2, separators=(',', ': '))
-        temp_path = '{}.tmp'.format(path)
-        self._pod.write_file(temp_path, output)
-        self._pod.move_file_to(temp_path, path)
+        fd, temp_path = tempfile.mkstemp()
+        with os.fdopen(fd, 'w') as fp:
+            fp.write(output)
+            real_path = self._pod.abs_path(path)
+            dir_name = os.path.dirname(real_path)
+            if not os.path.exists(dir_name):
+                os.makedirs(dir_name)
+            shutil.copyfile(temp_path, real_path)
```

Comment:
I actually think what you had before was more correct (`os.rename`?). `copyfile` will transfer bytes from one location to another, so afaik you'll still have the issue with one process reading in the middle of another process writing. I think `os.rename` will just update a pointer location which doesn't have the bytes being copied and will make it less likely for there to be conflicts.

Reply:
Also you should move the stuff after the `fp.write(output)` line outside of the `with` statement.

Reply:
OK - I think this is in its final form now. I was using shutil before because I thought mkstemp deleted the file outside the with statement. That was wrong (the file sticks around), so I can use os.rename.